lib/mysql_framework/sql_condition.rb

```diff
@@ -12,7 +12,11 @@ def initialize(column:, comparison:, value: nil)
       @column = column
       @comparison = comparison
 
-      raise ArgumentError, "Cannot set value when comparison is #{comparison}" if nil_comparison? && value != nil
+      if nil_comparison?
+        raise ArgumentError, "Cannot set value when comparison is #{comparison}" if value != nil
+      else
+        raise ArgumentError, "Comparison of #{comparison} requires value to be not nil" if value.nil?
```

Comment:
can we refactor this in some way that make it more obvious what we are validating?  The reason is because it looks like an if/else and at a quick first glance it looked like it was going to raise in either condition.  But really it is an if/if/else/if.

Reply:
possibly something like
```ruby
  ...
  validate
  ...
  @value = value
end

def validate
  raise ArgumentError, "Cannot set value when comparison is #{comparison}" unless valid_null_condition?
  raise ArgumentError, "Comparison of #{comparison} requires value to be not nil" unless valid_value?
end

def valid_null_condition?
  nil_comparison? == true && value.nil?
end

def valid_value?
  nil_comparison? == false && value != nil
end
```